### DISCUSSION TITLE
Application continues to run when unhandled exception is thrown in ReactiveCommand handler

### DISCUSSION THREAD

**Discussion by yudindm** [AvaloniaUI/Avalonia#15919] [Q&A]
Hi!

As I understand the Avalonia and Reactive UI documentation when unhandled exception is thrown while executing ReactiveCommand an application should terminate. But this does not happen in my test application.

The test application was created from Avalonia MVVM template.
Then I added ReactiveCommand to MainWindowViewModel which just throws exception and Button to invoke it in MainWindow.axaml.

```
// in MainWindowViewModel class
    public ReactiveCommand<Unit, Unit> ThrowExceptionCommand { get; }

// in MainWindowViewModel constructor
ThrowExceptionCommand = ReactiveCommand.Create(() =>
        {
            throw new Exception("Exception should terminate app.");
        });

// in MainWindow.axaml
    <StackPanel>
        <TextBlock Text="{Binding Greeting}"/>
        <Button Content="Throw Exception Command" Command="{Binding ThrowExceptionCommand}"/>
    </StackPanel>
```

When I run the application and clicked the Throw Exception Command button application did not terminate as I expected.
Any ideas what i did wrong?

**Comment by stevemonaco** [comment]
~~I created a new Avalonia MVVM App with ReactiveUI and I can't repro in VS -- I still get exceptions. Using the same C# / XAML. Check your IDE config for exceptions, maybe? Also please list which IDE you're using.~~

![image](https://github.com/AvaloniaUI/Avalonia/assets/1782158/cb78a10c-a32e-419a-af5d-38b0e2742a43)

**Reply by yudindm** [reply]
What happens if you press green "resume" button in the exception window or run application outside the IDE?
In my case the application continues to run which is not expected behavior if I correctly understand [this document](https://www.reactiveui.net/docs/handbook/default-exception-handler) 

> The default behaviour of ReactiveUI is to crash the application with whenever an object that has a ThrownExceptions property doesn't have a subscription.

**Reply by stevemonaco** [reply]
Ok, I understand. I can repro the app not exiting when this exception is thrown. The exception is getting swallowed somewhere and I'm not sure where.

Even when using a custom exception handler, throwing directly or scheduling a throw doesn't seem to exit. Avalonia has its own integration with RxUI, so maybe something is catching and not re-throwing...or maybe the exception isn't being marshaled.

**Reply by yudindm** [reply]
I feel this is a bug so I'v created the issue #16017
